arkouda/alignment.py

```diff
@@ -338,7 +338,7 @@ def search_intervals(vals, intervals, assume_unique=False):
         raise ValueError("intervals must be 2-tuple of (lower_bound_inclusive, upper_bounds_exclusive)")
 
     def check_numeric(x):
-        if not (isinstance(x, pdarray) and x.dtype in (akint64, akfloat64)):
+        if not (isinstance(x, pdarray) and x.dtype in (akint64, akuint64)):
```

Comment:
In the PR description it says it was not functional for `float64`. I think certain uses of `float64` are ok. but we do not support set operations on `float64` because of precision and instability reasons, unless we quantized/histogram first. Maybe some functions should allow this for boundaries... I'd need a better explanation. Maybe we should discuss.

Reply:
The issue with float64 is that `ak.pdarray._get_grouping_keys` does not support it. This results in `search_intervals` failing when called on floats64. If we update `ak.pdarray._get_grouping_keys` to allow for a float64, it enables float64 functionality for most of the setops functions and groupby. On the surface, they appear to work, but are slightly different in ouput (at least format) from numpy. Thus, I think we would need to do some more to get that working properly. I can certainly do that if it is the route we want to go. With that said, float64 will work if `assume_unique=true` because it does not run the `GroupBy`.  

`check_numeric` was not actually the issue, but I removed `akfloat64` since the containing function was not fully functional with that data type.

Let me know how we want to proceed. I can update setops to work for float64, but I would think that should be in a separate issue/PR.

Reply:
@reuster986 - I made the updates as requested. For the time being, I just set up the dtype handling to raise an exception so I could go take a look at the 2 bugs you put in marked important and the casting. If we prefer to try the cast and then error, I can update that.